### PR TITLE
feat: add paginated /names/search/:prefix endpoint

### PR DESCRIPTION
**Pull request by sborrazas**
Previously, this endpoint returned all results that matched on a
single page, potentially causing security concerns.

### CODE REVIEW

File: lib/ae_mdw_web/controllers/name_controller.ex [hunk 1]
```diff
@@ -229,28 +234,40 @@ defmodule AeMdwWeb.NameController do
...
-    do: do_auctions_stream(validate_params!(params), expand?(params))
+  def search(%Conn{request_path: path, assigns: assigns, query_string: query_string} = conn, %{
+        "prefix" => prefix
+      }) do
+    lifecycles =
+      query_string
+      |> URI.query_decoder()
+      |> Enum.filter(&match?({"only", _lifecycle}, &1))
+      |> Enum.map(fn {"only", lifecycle} -> lifecycle end)
+      |> Enum.filter(&Map.get(@lifecycles_map, &1))
```
Comment (jyeshe) [inline lib/ae_mdw_web/controllers/name_controller.ex:245]: maybe join this filter with previous one

**Review by jyeshe — COMMENTED** [review]
Looks good and test passes. Only one optional refactor opportunity.

**Comment by thepiwo** [comment]
@jyeshe @sborrazas do we want to merge those to a common branch for breaking changes we will accumulate?

**Review by thepiwo — APPROVED** [review]

**Review by thepiwo — APPROVED** [review]

File: lib/ae_mdw_web/controllers/name_controller.ex (hunk 1 above)
Comment (sborrazas) [reply lib/ae_mdw_web/controllers/name_controller.ex]: Done

**Review by jyeshe — APPROVED** [review]

**Pull request merged**